chapters/classes.tex

```diff
@@ -276,36 +276,127 @@ \subsubsection{Prefix Rules}\label{prefix-rules}
   function.
 \end{itemize}
 
-\subsection{Acyclic Bindings of Constants and Parameters}\label{acyclic-bindings-of-constants-and-parameters}
+\subsection{Acyclic Bindings of Constants and Structural Parameters}\label{acyclic-bindings-of-constants-and-parameters}
 
-The unexpanded binding equations for parameters and constants in the translated model must be acyclic after flattening; except that cycles are allowed if the cycles disappear when evaluating parameters having annotation \lstinline!Evaluate = true! that are not part of the cycle.
-Thus it is not possible to introduce equations for parameters by cyclic dependencies.
+The unexpanded binding equations for constants in the translated model must be acyclic after flattening.
+Thus it is not possible to introduce equations for constants by cyclic dependencies.
 
-\begin{nonnormative}
-There is no exception for parameters with \lstinline!fixed = false!, despite the fact that such parameters are generally allowed to be initialized from systems of dependent equations.
-% Saying "binding equation" below even though it's called a "declaration equation", for consistency with the normative paragraph above.
-However, a parameter with \lstinline!fixed = false! can use an initial equation instead of a binding equation, allowing for cyclic dependencies.
-\end{nonnormative}
+Structural parameters, on the other hand, cannot have cyclic dependencies by definition (see \cref{component-variability-prefixes-discrete-parameter-constant}).
+Since variability rules do not allow constants to depend on structural parameters, this means that constants and structural parameters share the property that their declaration equations can be solved one at a time with the natural causality.
 
 \begin{example}
 \begin{lstlisting}[language=modelica]
 constant Real p = 2 * q;
 constant Real q = sin(p); // Illegal since p = 2 * q, q = sin(p) are cyclical
+\end{lstlisting}
+\end{example}
+
+\subsection{Component Variability Prefixes discrete, parameter, constant}\label{component-variability-prefixes-discrete-parameter-constant}
+
+The prefixes \lstinline!discrete!, \lstinline!parameter!, \lstinline!constant! of a component declaration
+are called variability prefixes and define in which situation the
+variable values of a component are initialized (see \cref{events-and-synchronization} and
+\cref{initialization-initial-equation-and-initial-algorithm}) and when they are changed in transient analysis (= solution
+of initial value problem of the hybrid DAE):
+\begin{itemize}
+\item
+  A variable \lstinline!vc! declared with \lstinline!constant! prefix remains constant during transient analysis, with a value that is unaffected by the initialization problem (i.e., determined during translation).
+\item
+  % TODO: Add "structural parameter" to index.
+  A variable \lstinline!vc! is called a \firstuse{structural parameter} if all of the following applies:
```

Comment:
Good question.  I guess you are right about _n_, since _c_ was already taken for _constant_.

I fixed it so they at least have different names.  Personally, I don't really see the point of using different names at all.  Calling them all `v` or `x` allows the reader to concentrate on the definitions instead of thinking about what information the variable names are trying to convey.

Reply:
This conversation is very old and outdated in view of the last design meeting discussions.  Resolving.